.travis.yml

```diff
@@ -46,9 +38,11 @@ jobs:
       dist: xenial  # Travis won't install py37 on trusty
       install:
         - pip install "virtualenv<20.0.0" tox  # Bug https://github.com/pypa/virtualenv/issues/1551
+      before_script:
+        - source travis/scala_classpath_for_python.sh
       script:
-        - sbt +compile
-        - tox -c python/tox.ini -e py37 -v
+        - echo $SCALA_CLASS_PATH
+        - SCALA_CLASS_PATH=$SCALA_CLASS_PATH tox -c python/tox.ini -e py37 -v
```

Comment:
Sure, I'll document the steps in a comment in the next revision. 

Summing up, I just run this once:
`source travis/scala_classpath_for_python.sh`

It loads the classpath string into SCALA_CLASS_PATH which is then passed through to Python so that the classpath is available to the sparkContext for all test runs. 

For tests you can run:
`tox -c python/tox.ini -e py37 -v`
or also you can run any test singularly, the environemnt variable is always available.

(If you check the tox.ini file under `passenv`, I'm forwarding SCALA_CLASS_PATH there.)

Reply:
Thanks.

Also I’m wondering specifically if there’s some handy way to make the env variable available in an IDE like PyCharm. Especially if the classpath changes because of something, would I need to copy the new env var to my IDE manually? Previously everything was bound to the sbt compilation so it was naturally always up to date. I understand the reasons why you’re changing this, but still, I hope there would be some clean way to setup IDE without too much manual maintenance.

Reply:
So there's a few ways I would do this in PyCharm. 
1. set an env. variable in vanilla PyCharm, explanation here: https://stackoverflow.com/a/42708480/2544874
2. You can also use this JetBrains plugin: https://plugins.jetbrains.com/plugin/7861-envfile
it basically lets you load environment variables from a file. When I use that plugin I ensure I generate an hidden .env_variables.env file as it's best to avoid committing  workflow related files and procedures into the project.
3. Otherwise you can run from command line (this is my preferred method cause it lets me test things cross-platform, including what happens on Travis) :
```
source travis/scala_classpath_for_python.sh (this only once in the terminal)
tox -c python/tox.ini -e py37 -v
```

I can't think of a circumstance where the classpath would change "often". If it does, you can `source` again, or you could have your own script to regenerate the .env file in option 2) or reset option 1)

Reply:
Thanks for those tips – the env file plugin can make this convenient enough at least.

Do you think some of those tips could make their way into a README?

> I can't think of a circumstance where the classpath would change "often"

Agreed. The point is though, I'd like it to be convenient when it happens. If I pull changes from repo that includes some change to the classpath, I then run sbt compile which is standard, but for python there's a unofficial step to remember / look up.

If nothing else, should the README include this that code snippet:

```
source travis/scala_classpath_for_python.sh (this only once in the terminal)
tox -c python/tox.ini -e py37 -v
```

?

Reply:
That's reasonable!

I think Makefile have this job generally, MLeap doesn't have a top-level Makefile but uses Travis, so I hoped that reading Travis should be enough, but it's probably very helpful to add all of this in the python specific Makefile too as many devs refer to Makefiles as a first thing.

If you feel strongly about a README, I'll add it in there too